BGL/test/BGL/test_Collapse_edge.cpp

```diff
@@ -0,0 +1,226 @@
+#include "test_Prefix.h"
+#include <boost/range/distance.hpp>
+#include <CGAL/boost/graph/Euler_operations.h>
+
+template < typename Mesh>
+typename boost::graph_traits<Mesh>::
+halfedge_descriptor find_halfedge(double x1, double y1,
+                                  double x2, double y2,
+                                  Mesh& m,
+                                  bool is_border = false)
+{
+  typedef typename boost::property_map<Mesh, CGAL::vertex_point_t>::type VPMAP;
+  typedef typename boost::property_traits<VPMAP>::value_type Point;
+
+  typedef typename boost::graph_traits<Mesh>::halfedge_descriptor halfedge_descriptor;
+  VPMAP vpmap = get(CGAL::vertex_point, m);
+  BOOST_FOREACH(halfedge_descriptor h, halfedges(m))
+  {
+    if(get(vpmap, source(h, m)) == Point(x1,y1,0)
+       && get(vpmap, target(h, m)) == Point(x2,y2,0))
+    {
+      if(is_border == CGAL::is_border(h, m))
+        return h;
+      else
+        return opposite(h, m);
+    }
+  }
+  return boost::graph_traits<Mesh>::null_halfedge();
+}
+
+template <typename Mesh>
+void
+collapse_edge_test()
```

Comment:
everywhere in this test, I suggest to call `h` or `he` the `halfedge_descriptor`, instead of `e`, which seems more to be an edge.

The reason is that when I read `e = find_halfedge(...)` or `opposite(e, g)`, it seems to be a bad use of the types.

This remark is for future debugging